### PR TITLE
Adding a "mode" field to the MarginCalcResponse

### PR DESCRIPTION
**Pull request by andreiruse**
This field represents the mode the calculation has ran in. 
The valid modes are defined in `MarginCalcMode`

### CODE REVIEW

File: modules/margin/src/main/java/com/opengamma/sdk/margin/MarginCalcMode.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+   * Runs the margin calculation in Spot mode.
+   */
+  SPOT,
+
+  /**
+   * Runs the margin calculation in Forward Approximation mode.
+   */
+  FORWARD
+
+}
```
Comment (jodastephen) [inline modules/margin/src/main/java/com/opengamma/sdk/margin/MarginCalcMode.java]: New line EOF

File: modules/margin/src/main/java/com/opengamma/sdk/margin/MarginCalcMode.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * Copyright (C) 2018 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.sdk.margin;
+
+/**
+ * The mode in which the calculation will be executed.
+ */
+public enum MarginCalcMode {
```
Comment (jodastephen) [inline modules/margin/src/main/java/com/opengamma/sdk/margin/MarginCalcMode.java:11]: Shouldn't this be on the input as well?

Comment (andreiruse) [reply modules/margin/src/main/java/com/opengamma/sdk/margin/MarginCalcMode.java:11]: Good point - added

File: modules/margin/src/main/java/com/opengamma/sdk/margin/MarginCalcRequest.java [hunk 3]
```diff
@@ -39,6 +39,11 @@
    */
   @PropertyDefinition(validate = "notNull")
   private final MarginCalcRequestType type;
+  /**
+   * The mode of the calculation, defaulted to {@code SPOT}.
```
Comment (jodastephen) [inline modules/margin/src/main/java/com/opengamma/sdk/margin/MarginCalcRequest.java:43]: I'd expect to see `@ImmutableDefaults`

File: modules/margin/src/main/java/com/opengamma/sdk/margin/MarginCalcRequest.java [hunk 4]
```diff
@@ -109,6 +116,39 @@ public static MarginCalcRequest of(
...
    *
+   * @param mode  the mode of the calculation
+   * @param valuationDate  the valuation date
+   * @param reportingCurrency  the reporting currency
+   * @param portfolioData  the portfolio data, which can be {@link PortfolioDataFile} or Strata trades
+   * @param fpmlPartySelectionRegex  the regular expression used to select the party in any FpML input
+   * @return the request
+   */
+  public static MarginCalcRequest of(
+      MarginCalcMode mode,
```
Comment (jodastephen) [inline modules/margin/src/main/java/com/opengamma/sdk/margin/MarginCalcRequest.java]: This should be after the reporting currency - its not as important as date or currency

**Review by jodastephen — APPROVED** [review]

**Pull request merged**